docs/integrations/sources/s3.md

```diff
@@ -152,66 +172,105 @@ Please note, the S3 Source connector used to infer schemas from all the availabl
-- `columns` : If not None, only these columns will be read from the file.
-- `batch_size` : Maximum number of records per batch. Batches may be smaller if there aren’t enough rows in the file.
+- **Selected Columns**: If you only want to sync a subset of the columns from the file(s), enter the desired columns here as a comma-delimited list. Leave this field empty to sync all columns.
+- **Record Batch Size**: Sets the maximum number of records per batch. Batches may be smaller if there aren’t enough rows in the file. This option can help avoid out-of-memory errors if your data is particularly wide. Set to `65536` by default.
+- **Buffer Size**: If set to a positive number, read buffering is performed during the deserializing of individual column chunks. Otherwise I/O calls are unbuffered. Set to `2` by default.
 
-You can find details on [here](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetFile.html#pyarrow.parquet.ParquetFile.iter_batches).
+For more information on these fields, please refer to the [Apache documentation](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetFile.html#pyarrow.parquet.ParquetFile.iter_batches).
 
 ### Avro
 
-The avro parser uses [fastavro](https://fastavro.readthedocs.io/en/latest/). Currently, no additional options are supported.
+The Avro parser uses the [Fastavro library](https://fastavro.readthedocs.io/en/latest/). Currently, no additional options are supported.
+
+### JSONL
+
+The JSONL parser uses the PyArrow library, which only supports the line-delimited JSON format. For more detailed info, please refer to the [official docs](https://arrow.apache.org/docs/python/json.html).
 
-### Jsonl
+- **Allow newlines in values**: While JSONL typically has each JSON object on a separate line, there are cases where newline characters may appear within JSON values, such as multiline strings. This option enables the parser to correctly interpret and treat newline characters within values. Please note that setting this option to True may affect performance. Set to False by default.
 
-The Jsonl parser uses pyarrow hence,only the line-delimited JSON format is supported.For more detailed info, please refer to the [docs](https://arrow.apache.org/docs/python/generated/pyarrow.json.read_json.html)
+- **Unexpected field behavior**: This parameter determines how any JSON fields outside of the explicit schema (if defined) are handled. Possible behaviors include:
+
+  - `ignore`: Unexpected JSON fields are ignored.
+  - `error`: Error out on unexpected JSON fields.
+  - `infer`: Unexpected JSON fields are type-inferred and included in the output.
+
+Set to `infer` by default.
+
+- **Block Size**: This sets the number of bytes to process in memory at a time while reading files. The default value of `10000` is usually suitable, but if your table is particularly wide \(lots of columns / data in fields is large\), increasing this might help avoid schema detection failures.
```

Comment:
```suggestion
- **Block Size**: This sets the number of bytes to process in memory at a time while reading files. The default value of `10000` is usually suitable, but if your files are particularly wide \(lots of columns or the values in the columns are particularly large), increasing this might help avoid schema detection failures.
```